### PR TITLE
feat: sign macos binary with travis

### PR DESCRIPTION
**Pull request by dimabru**
* Added ability to sign and notarize cli mac executable using travis during cicd
* Signed release will work in `staging` and `main` branches

### CODE REVIEW

**Review by myishay — CHANGES_REQUESTED** [review]
coooool
please add internal release support

File: .travis.yml [hunk 1]
```diff
@@ -1,35 +1,33 @@
...
-  on:
-    all_branches: true
-    condition: $TRAVIS_BRANCH =~ ^(staging)$
-
+  - provider: script
+    skip_cleanup: true
+    script: bash ./internal/release/deploy.sh
+    on:
+      all_branches: true
+      condition: $TRAVIS_BRANCH =~ ^(staging|DAT-3096-cicd-signing)$
```
Comment (alexfedin) [inline .travis.yml]: we should remove this before merging

File: internal/release/deploy.sh [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+security create-keychain -p test buildagent.keychain
+security default-keychain -s buildagent.keychain
+security unlock-keychain -p test buildagent.keychain
+security list-keychains -d user -s buildagent.keychain
+security import DatreeCli.p12 -k buildagent.keychain -P $P12_PASSWORD -T /usr/bin/codesign
+security set-key-partition-list -S "apple-tool:,apple:" -s -k test buildagent.keychain
+security find-identity -v
+curl -sL https://git.io/goreleaser | VERSION=v$GORELEASER_VERSION bash
+
+if [ $TRAVIS_BRANCH == "DAT-3096-cicd-signing" ]; then
```
Comment (alexfedin) [inline internal/release/deploy.sh]: again, don't forget to remove this before a merge

**Review by alexfedin — DISMISSED** [review]

File: .travis.yml (hunk 1 above)
Comment (dimabru) [reply .travis.yml]: removed

File: internal/release/deploy.sh (hunk 2 above)
Comment (dimabru) [reply internal/release/deploy.sh]: removed

**Pull request merged**